changelog/2024-02-23T16_56_56+01_00_fix2628

```diff
@@ -0,0 +1 @@
+FIXED: Clash errors our in netlist-generation stage out when a polymorphic function is applied to type X in one alternative of a case-statement and applied to a new-type wrapper of type X in a different alternative. See [#2828](https://github.com/clash-lang/clash-compiler/issues/2628)
```

Comment:
Also, "netlist" and "generation" should probably not be hyphenated.

Reply:
Neither should _case-statement_ or _new-type_

```suggestion
FIXED: Clash no longer errors out in netlist generation stage when a polymorphic function is applied to type X in one alternative of a case statement and applied to a newtype wrapper of type X in a different alternative. See [#2828](https://github.com/clash-lang/clash-compiler/issues/2628)
```

Also,  I am to understand that Clash now no longer errors out? That's not what it said: Fixed: Clash errors out. Without emphasis, I would interpret that as "Clash didn't use to error out, this is now fixed so it properly errors out." However, I believe that is the wrong interpretation. So I changed it to "no longer errors out" to emphasise this. If instead it's meant that it now errors out, it should say "Clash now errors out". The old phrasing was just too ambiguous.

Reply:
We currently mix: `FIXED: {issue subject}` and `FIXED: {changelog styled entry}`, which is confusing. We should pick one, yeah.

Reply:
If we make sure it can only be interpreted one way (_now_, _no longer_), people won't have to analyse the other entries in the changelog to see which style we use. While it may not be pretty if we mix the two styles you mention, we can at least prevent ambiguity by taking an adversarial approach to reading our own changelog entries: can I actually read it in a different way? If so, rephrase.

Reply:
Agreed, I like the changelog-style way.

Reply:
Neither the title nor the content of https://github.com/clash-lang/clash-compiler/issues/2628 is really descriptive of the defect.

Anyhow, I feel, the changelog should be in the form:

```
FIXED: the old defect that is now fixed
CHANGED: old situation, new situation
ADDED: the thing being added
```

That is, I understand the `FIXED` as "bug remedied".

Reply:
Well to me that certainly does not imply it in a changelog; in practice I see both used. So please phrase it unambiguously. As I indicated, I initially read it wrong. This is not a "somebody could read it the wrong way", it's an "I actually read it the wrong way".

I've definitely phrased more loosely than your template when writing our own changelog in the past, by the way.

Reply:
As I said, I don't really care from which perspective you phrase as long as it is difficult to misinterpret. However...

```
FIXED: the old defect that is now fixed
```
```
FIXED: Clash errors out in the netlist generation stage when ...
```
Hmmm... can you give me a proper sentence where the verb that describes the issue is in present tense? Because I have trouble finding one. If somebody told me "I've mended your vase, it is in a thousand pieces" I'd definitely give them a funny look. I would expect them to use past tense. So I'd expect "Clash errored out" here.

Since this is a single sentence with 36 words and a very non-trivial sentence structure, I do feel that just relying on verb inflection is too little. There's already too much mental processing happening for the reader, a mistake is easily made in interpretation. I'd rather be clunky and clear than elegant and vague.

Reply:
But this changelog entry will become part of a list:
```
FIXED:
* defect description 1
* defect description 2
* defect description 3
* etc.
```
Perhaps instead of `FIXED`, it should say `BUGS FIXED`.

I can chance the description so that it says:
```
Clash falsely errors out during netlist generation when a polymorphic function is applied to type X in one alternative of a case-statement and applied to a newtype wrapper of type X in a different alternative. See [#2828](https://github.com/clash-lang/clash-compiler/issues/2628
```

Reply:
But why do you keep insisting on the present tense? By the time people read the changelog, the behaviour will be in the past. Can you give me a proper sentence where the problem is described in the present tense? This present tense only serves to confuse. Not only is it wrong, it is confusing, because it is wrong. We're trying to communicate to our users here, not to make some point.

Reply:
It is confusing, and therefore it gets a NAK from me. I don't care how you phrase it as long as it is not confusing. This is confusing. The intent is to communicate clearly.